components/tools/OmeroPy/src/omero/plugins/sessions.py

```diff
@@ -892,12 +899,18 @@ def _get_server(self, store, name, port):
     def _get_username(self, defuser):
         if defuser is None:
             defuser = get_user("root")
+        self._require_tty("cannot request username")
         rv = self.ctx.input("Username: [%s]" % defuser)
         if not rv:
             return defuser
         else:
             return rv
 
+    def _require_tty(self, msg):
+        if sys.stdin.isatty():
+            return
+        self.ctx.die(564, "stdin is not a terminal: %s" % msg)
```

Comment:
No, though in future it'd be nice if these were defined somewhere so we could use them.